tests/profiles/bigquery/test_bq_service_account_keyfile_dict.py

```diff
@@ -64,8 +64,8 @@ def test_connection_claiming_succeeds(mock_bigquery_conn_with_dict: Connection):
 
 
 def test_connection_claiming_fails(mock_bigquery_conn_with_dict: Connection):
-    # Remove the `dataset` key, which is mandatory
```

Comment:
Do we still have unit tests covering that if users were defining the Connection `extras.dataset`, it would be used?

Do we have a unittest covering that if users do not set `extras.dataset`, that we're not raising an exception?

Reply:
Test where we set dataset https://github.com/astronomer/astronomer-cosmos/blob/main/tests/profiles/bigquery/test_bq_service_account_keyfile_dict.py#L73

I have added a test where we set dataset and it does not raise error

Reply:
Thank you very much, @pankajastro !